scenario_player/definition.py

```diff
@@ -23,19 +23,19 @@ class ScenarioDefinition:
     def __init__(
         self, yaml_path: pathlib.Path, data_path: pathlib.Path, environment: EnvironmentConfig
     ) -> None:
+        self._scenario_dir = None
         self.path = yaml_path
         # Use the scenario file as jinja template and only parse the yaml, afterwards.
         with yaml_path.open() as f:
             yaml_template = jinja2.Template(f.read(), undefined=jinja2.StrictUndefined)
             rendered_yaml = yaml_template.render(**asdict(environment))
             self._loaded = yaml.safe_load(rendered_yaml)
 
-        self._scenario_dir = None
-        self.token = TokenConfig(self._loaded, data_path.joinpath("token.info"))
-        deploy_token = self.token.address is None
-        self.nodes = NodesConfig(self._loaded, environment="development" if deploy_token else None)
         self.settings = SettingsConfig(self._loaded, environment)
         self.settings.sp_root_dir = data_path
+        self.token = TokenConfig(self._loaded, self.scenario_dir.joinpath("token.info"))
```

Comment:
How can `self._scenario_dir` not be `None` here?

Reply:
`scenario_dir` (without the underscore) is a property. It reads it from the settings.